### PR TITLE
PR for Workshop 3 by Rakesh Kumar Singh

### PR DESCRIPTION
**Pull request by rakeshkumar1019**

### CODE REVIEW

**Review by Gabez0r — COMMENTED** [review]
Looking good. You might want to review the [ex3: commenting cancel test case](https://github.com/nuxeo/nuxeo-web-ui/pull/1495/commits/df254524403c500fdd20c3862092e49225aaea29) commit, which seems to have the wrong message.

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 1]
```diff
@@ -125,14 +149,32 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
...
+                      class="current breadcrumb-item breadcrumb-item-current"
+                      aria-current="page"
+                      title="[[_title(document)]]"
+                    >
+                      [[_title(document)]]
+                    </a>
+                  
+                        <span class="doc-uid">
+                        [[_documentUID(document)]]
+                      </span>
```
Comment (Gabez0r) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js]: The indentation seems to be off here.

**Review by Gabez0r — COMMENTED** [review]
Good job. Please check comments under https://github.com/nuxeo/nuxeo-web-ui/pull/1454#pullrequestreview-1015043759.

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 2]
```diff
@@ -125,14 +149,29 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
...
+                      href$="[[urlFor(document)]]"
+                      class="current breadcrumb-item breadcrumb-item-current"
+                      aria-current="page"
+                      title="[[_title(document)]]"
+                    >
+                      [[_title(document)]]
+                    </a>
+                       <span class="doc-uid">
+                        [[_documentUID(document)]]
+                       </span>
```
Comment (Gabez0r) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js]: Indentation seems to be a bit off here.

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 3]
```diff
@@ -125,14 +149,29 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
...
+                </div>
+                <template is="dom-if" if="[[_isTrashed(document)]]">
+                  <span class="trash-icon-parent">
+                   <paper-icon-button icon="nuxeo:delete"
```
Comment (Gabez0r) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js]: Note that there's no good reason to user a button here if we are not expecting the user to click on it. We should use the [iron-icon](https://www.webcomponents.org/element/@polymer/iron-icon) instead, which already exposes [a couple of CSS variables to set width and height](https://github.com/PolymerElements/iron-icon/blob/master/iron-icon.js#L79-L80).

**Review by Gabez0r — COMMENTED** [review]
Good work. Please see comments for:
- development: https://github.com/nuxeo/nuxeo-web-ui/pull/1454#pullrequestreview-1024500921
- functional tests: https://github.com/nuxeo/nuxeo-web-ui/pull/1491#pullrequestreview-1024516164

File: elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+import '@nuxeo/nuxeo-ui-elements/widgets/nuxeo-tooltip.js';
+import '../document/nuxeo-document-form-layout.js';
+import '@polymer/paper-button/paper-button.js';
+
+/**
+`nuxeo-document-parent-inspector-button`
+@group Nuxeo UI
+@element nuxeo-document-parent-inspector-button
+*/
+Polymer({
```
Comment (richardsd) [inline elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js]: Please refer to Gabriel's comment here https://github.com/nuxeo/nuxeo-web-ui/pull/1454#discussion_r910797179.

File: elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+        <div class="parentscroll">
+          <div class="scrollable">
+            <div>
+              <h1>Parent Inspector</h1>
+              <div>
+                <p>This is the parent inspector. It displays information about the parent of this document.</p>
+                <table class="table">
+                  <tr>
+                    <td>Tile:</td>
+                    <td>[[document.title]]</td>
```
Comment (richardsd) [inline elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js]: Here you are displaying information about the current document, not the parent.

File: elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+  _openDialog() {
+    this.dialog.open();
+  },
+
+  _closeDialog() {
+    this.dialog.close();
+  },
+
+  _computeLabel() {
+    const x = this.i18n('parentInspectorButton.tooltip');
```
Comment (richardsd) [inline elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js]: No need to save this in a variable if you are not doing anything with it.
```suggestion
return this.i18n('parentInspectorButton.tooltip');
```

Comment (richardsd) [reply elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js]: Even better you could use just `i18n('parentInspectorButton.tooltip')` directly in the 

```
<nuxeo-tooltip>[[i18n('parentInspectorButton.tooltip')]]</nuxeo-tooltip>
```

File: elements/nuxeo-web-ui-bundle.html [hunk 7]
```diff
@@ -231,6 +231,16 @@
 
 <!-- DOCUMENT ACTIONS -->
 
+<nuxeo-slot-content name="parentInspectorDocumentAction" slot="DOCUMENT_ACTIONS" order="5">
+  <template>
+    <nuxeo-filter user="[[user]]" expression="user.isAdministrator || this.isMember(user, 'powerusers')">
```
Comment (richardsd) [inline elements/nuxeo-web-ui-bundle.html]: As an alternative solution, please check https://github.com/nuxeo/nuxeo-elements/blob/maintenance-3.0.x/ui/nuxeo-filter.js#L167. nuxeo-filter has a group property that can be use to check user membership to groups.

File: i18n/messages-ar.json [hunk 8]
```diff
@@ -983,6 +983,7 @@
   "label.dublincore.state": "State",
   "label.dublincore.subjects": "Subjects",
   "label.dublincore.title": "Title",
+  "label.dublincore.source": "Source",
```
Comment (richardsd) [inline i18n/messages-ar.json:986]: No need to manually add the label to all the `messages-*.json` files. You only need to add it to `messages.json`, then the translation process will happen at Crowdin. Please read the wiki page about the translation process for further details about this.

File: packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import pause from 'webdriverio/build/commands/browser/pause';
+
+Then('I can open current document information', function() {
+  this.ui.browser.parentInspectorButton.waitForVisible();
+  this.ui.browser.parentInspectorButton.click();
+  this.ui.browser.parentInspectorDialog.waitForVisible();
+});
+
+Then('I can see dialog showing the information for a given document', function() {
+  this.ui.browser.parentInspectorDataTitle === 'Title:';
```
Comment (richardsd) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js]: You should instead validate that the fields are visible in the parent inspect (similar to what you doing in the `I can see dialog showing some information for a given document` step to validate that some fields are not displayed).

File: packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  this.ui.browser.parentInspectorDataPath === 'Path:';
+  this.ui.browser.parentInspectorDataUID === 'UID:';
+  this.ui.browser.parentInspectorDataFacets === 'Facets:';
+  this.ui.browser.parentInspectorDataSchema === 'Schemas:';
+});
+
+Then('I can see dialog showing some information for a given document', function() {
```
Comment (richardsd) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js]: See my comment above.

**Review by richardsd — COMMENTED** [review]
It is looking good 👍 And the PR checks are passing 🙂

**Review by Gabez0r — COMMENTED** [review]
Well done. Please see comments here: https://github.com/nuxeo/nuxeo-web-ui/pull/1491#pullrequestreview-1027608150.

**Review by Gabez0r — COMMENTED** [review]
Well done! Please see comments here: https://github.com/nuxeo/nuxeo-web-ui/pull/1491#pullrequestreview-1028279090

File: packages/nuxeo-web-ui-ftest/features/step_definitions/login.js [hunk 11]
```diff
@@ -16,6 +16,19 @@ Given('user {string} exists in group {string}', (username, group) =>
   }),
 );
 
+Given('user {string} exists in groups {string} and {string}', (username, group1, group2) =>
```
Comment (Gabez0r) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/login.js]: Please see comment https://github.com/nuxeo/nuxeo-web-ui/pull/1491#discussion_r913777257

**Review by Gabez0r — COMMENTED** [review]
Good job.

**Pull request closed without merging**